### PR TITLE
Resource: nginx_install

### PR DESCRIPTION
**Pull request by fernandoaleman**
# Resource: nginx_install

- Update kitchen to use chef solo
- Support only OS with systemd
- Drop support for Amazon Linux, CentOS 6
- Add support for Amazon Linux 2
- Remove community_cookbook_releaser gem
- Move spec/libraries to spec/unit/libraries
- Bump ohai dependency to ~> 5.2
- Create nginx_install custom resource with source properties distro, repo, epel and passenger
- Create nginx_config custom resource
- Update nginx_site resource
- Update default-site.erb template
- Default recipe calls nginx_install resource

## Issues Resolved

- Addressing Issue #486 
- Can now install nginx via resource vs recipes.

## Check List

- [x] All tests pass. See TESTING.md for details
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable

## TODO

- Decide if we are going to keep the option to install nginx from `source`. There was a discussion about this in the slack channel, so just have to confirm.

- The kitchen.yml still includes the `source` and `modules` tests and those are currently failing, pending the decision to keep or remove the option to build from source.

This is my first time submitting a PR to the `sous-chef` group so I look forward to any feedback and suggestions on improving the PR.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,7 @@
-source 'https://rubygems.org'
+# This gemfile provides additional gems for testing and releasing this cookbook
```
[the remark this answers is not in the document]
Comment (fernandoaleman) [reply Gemfile]: @JohnRoesler I don't believe so. I can remove it.

**Comment by fernandoaleman** [comment]
@damacus Hello Dan. Was the consensus that we can indeed remove the option to install from `source` or do we still need that?

**Comment by damacus** [comment]
I think for simplicity we sohuld remove the source install, if someone asks for it again. We'll put it back. 

Does anyone here still use source install for installing mods?

Also @fernandoaleman can you rebase on master please? Looksl ike we've got a few conflicts

**Comment by damacus** [comment]
@fernandoaleman can you rebase on master please?

**Comment by fernandoaleman** [comment]
@damacus Yes, I'll do that now. I did a few updates recently but some things failed CI. I'll look into that as well.

**Comment by damacus** [comment]
Awesome. 

Most of those `delivery` failures looks like running `cookstyle -a` will fix them automatically, if you have the latest ChefDK

**Comment by fernandoaleman** [comment]
@damacus Ok, I've rebased master and updated the CHANGELOG. All tests are now passing. Can you please review when you have a moment please? It's a pretty simplified cookbook now.

**Comment by alhughes04** [comment]
I used version 9.0.0 of this cookbook to generate nginx configuration but not install the nginx service. I found this useful for quickly generating nginx configuration that could then be leveraged to stand up nginx in Docker. With the removal of the old recipes and creation of the new config resource is this still possible? Specifically, the config.rb resource notifies the nginx service to reload thus requiring the nginx service to be installed on the host. https://github.com/sous-chefs/nginx/blob/d3579b9034386d6f353e8210f1c1ecd374fb2bc4/resources/config.rb#L56

I can appreciate that this cookbook was designed for this purpose but it would be great if it was flexible enough to work with a Docker install as well. Let me know if I'm missing something. Thanks in advance.

**Comment by damacus** [comment]
@alhughes04 if you can open a specific issue for Docker with a clearer description of where the current master version doesn't work that'd be great.

**Pull request merged**